### PR TITLE
Add detail about self-healing

### PR DESCRIPTION
**Pull request by Gianluca755**
Added a detail about self-healing, I think for better clarity

### CODE REVIEW

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 1]
```diff
@@ -11,10 +11,11 @@ in the `Pending` [phase](#pod-phase), moving through `Running` if at least one
...
 `Failed` phases depending on whether any container in the Pod terminated in failure.
 
-Whilst a Pod is running, the kubelet is able to restart containers to handle some
-kind of faults. Within a Pod, Kubernetes tracks different container
+Within a Pod, Kubernetes tracks different container
 [states](#container-states) and determines what action to take to make the Pod
-healthy again.
+healthy again. Whilst a Pod is running, the kubelet is able to restart containers to
+handle some kind of faults. Pods are not restarted, they are terminated and respawned
+(recreated).
```
Comment (shannonxtreme) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:18]:
```suggestion
handle some kind of faults. The kubelet doesn't restart unhealthy _Pods_; it terminates and then recreates the Pods.
```

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 2]
```diff
@@ -38,11 +39,12 @@ If a {{< glossary_tooltip term_id="node" >}} dies, the Pods scheduled to that no
...
 Pods do not, by themselves, self-heal. If a Pod is scheduled to a
-{{< glossary_tooltip text="node" term_id="node" >}} that then fails, the Pod is deleted; likewise, a Pod won't
-survive an eviction due to a lack of resources or Node maintenance. Kubernetes uses a
-higher-level abstraction, called a
-{{< glossary_tooltip term_id="controller" text="controller" >}}, that handles the work of
-managing the relatively disposable Pod instances.
+{{< glossary_tooltip text="node" term_id="node" >}} that then fails, the Pod is
+deleted; likewise, a Pod won't survive an eviction due to a lack of resources or
+Node maintenance. What Pods can do is self-heal on a local node. In order to have
+self-healing across the cluster, Kubernetes uses a higher-level abstraction, called
```
Comment (shannonxtreme) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:45]: To me, self-healing explicitly says that the Pod controls its own health. I don't think that's accurate though, it's a controller than manages the Pod lifecycle and not the Pod itself that manages it. 

Maybe ask someone from @kubernetes/sig-node-pr-reviews or a more experienced docs reviewer for confirmation, but I feel like this muddles the clarity a teeny bit

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 3]
```diff
@@ -38,11 +39,12 @@ If a {{< glossary_tooltip term_id="node" >}} dies, the Pods scheduled to that no
...
 
 Pods do not, by themselves, self-heal. If a Pod is scheduled to a
-{{< glossary_tooltip text="node" term_id="node" >}} that then fails, the Pod is deleted; likewise, a Pod won't
-survive an eviction due to a lack of resources or Node maintenance. Kubernetes uses a
-higher-level abstraction, called a
-{{< glossary_tooltip term_id="controller" text="controller" >}}, that handles the work of
-managing the relatively disposable Pod instances.
+{{< glossary_tooltip text="node" term_id="node" >}} that then fails, the Pod is
+deleted; likewise, a Pod won't survive an eviction due to a lack of resources or
+Node maintenance. What Pods can do is self-heal on a local node. In order to have
```
Comment (shannonxtreme) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:44]:
```suggestion
Node maintenance. What Pods can do is self-heal on a node. In order to have
```

Comment (shannonxtreme) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:44]:
```suggestion
Node maintenance. What Pods can do is self-heal on a local node. To have
```

**Review by shannonxtreme — CHANGES_REQUESTED** [review]

**Comment by shannonxtreme** [comment]
Whoops, and please make sure that you sign the CLA!

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md:18]: The kubelet doesn't **recreate** failed Pods. The control plane sometimes does that though - specifically, when some higher level API such as a ReplicaSet decides that this should happen.

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md (hunk 3 above)
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:44]: > What Pods can do is self-heal on a local node.

This is not right. When a Pod fails, a controller might replace it. It's up to that controller and / or the control plane to decide what scheduling constraints to apply, and ultimately where to place that replacement pod.

For example, when a node fails, the replacements for the pods that were running there are likely to end up on a different, healthy node.

**Review by Arhell — COMMENTED** [review]
CLA info
https://github.com/kubernetes/community/blob/master/CLA.md

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md (hunk 1 above)
Comment (endocrimes) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md:18]: +1

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md (hunk 2 above)
Comment (endocrimes) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md:45]: yeah this muddies things a little for me too.

**Comment by sftim** [comment]
Ah yep, this needs a CLA before we review it!

**Pull request closed without merging**